bot/bot.py

```diff
@@ -223,81 +116,41 @@ def insert_item_into_filter_list_cache(self, item: Dict[str, str]) -> None:
-        # Use asyncio for DNS resolution instead of threads so threads aren't spammed.
-        self._resolver = aiohttp.AsyncResolver()
-
-        # Use AF_INET as its socket family to prevent HTTPS related problems both locally
-        # and in production.
-        self._connector = aiohttp.TCPConnector(
-            resolver=self._resolver,
-            family=socket.AF_INET,
+    async def setup_hook(self) -> None:
+        """Default Async initialisation method for Discord.py."""
+        self.api_client = APIClient(
+            constants.URLs.site_api_schema+constants.URLs.site_api,
+            constants.Keys.site_api,
+            connector=self._connector
         )
-
-        # Client.login() will call HTTPClient.static_login() which will create a session using
-        # this connector attribute.
-        self.http.connector = self._connector
-
-        self.http_session = aiohttp.ClientSession(connector=self._connector)
-        self.api_client = api.APIClient(connector=self._connector)
-
+        await super().setup_hook()
         if self.redis_session.closed:
             # If the RedisSession was somehow closed, we try to reconnect it
             # here. Normally, this shouldn't happen.
             await self.redis_session.connect()
 
-        try:
-            await self.ping_services()
-        except Exception as e:
-            raise StartupError(e)
-
         # Build the FilterList cache
         await self.cache_filter_list_data()
 
         await self.stats.create_socket()
-        await super().login(*args, **kwargs)
-
-    async def on_guild_available(self, guild: discord.Guild) -> None:
-        """
-        Set the internal guild available event when constants.Guild.id becomes available.
-
-        If the cache appears to still be empty (no members, no channels, or no roles), the event
-        will not be set.
-        """
-        if guild.id != constants.Guild.id:
-            return
-
-        if not guild.roles or not guild.members or not guild.channels:
-            msg = "Guild available event was dispatched but the cache appears to still be empty!"
-            log.warning(msg)
-
-            try:
-                webhook = await self.fetch_webhook(constants.Webhooks.dev_log)
-            except discord.HTTPException as e:
-                log.error(f"Failed to fetch webhook to send empty cache warning: status {e.status}")
-            else:
-                await webhook.send(f"<@&{constants.Roles.admin}> {msg}")
-
-            return
 
-        self._guild_available.set()
-
-    async def on_guild_unavailable(self, guild: discord.Guild) -> None:
-        """Clear the internal guild available event when constants.Guild.id becomes unavailable."""
-        if guild.id != constants.Guild.id:
-            return
-
-        self._guild_available.clear()
+        # Must be done here to avoid a circular import.
+        from bot.utils.extensions import EXTENSIONS
```

Comment:
Is this still necessary if bot core is being used? Also, I don't think you ever updated the extensions util to use bot core (I don't see that file modified in the diff).